src/System.ComponentModel.TypeConverter/tests/MemberDescriptorTests.cs

```diff
@@ -2,105 +2,1547 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections;
+using System.Collections.Generic;
 using System.Linq;
+using System.Reflection;
 using Xunit;
 
 namespace System.ComponentModel.Tests
 {
     public class MemberDescriptorTests
     {
+        [Theory]
+        [InlineData("  ")]
+        [InlineData("name")]
+        public void Ctor_String(string name)
+        {
+            var descriptor = new SubMemberDescriptor(name);
+            Assert.Empty(descriptor.Attributes);
+            Assert.Same(descriptor.Attributes, descriptor.Attributes);
+            Assert.Empty(descriptor.AttributeArray);
+            Assert.Same(descriptor.AttributeArray, descriptor.AttributeArray);
+            Assert.Equal("Misc", descriptor.Category);
+            Assert.Empty(descriptor.Description);
+            Assert.False(descriptor.DesignTimeOnly);
+            Assert.Same(name, descriptor.DisplayName);
+            Assert.True(descriptor.IsBrowsable);
+            Assert.Equal(name, descriptor.Name);
+            Assert.Equal(name.GetHashCode(), descriptor.NameHashCode);
+        }
+
+        public static IEnumerable<object[]> Ctor_String_Attributes_TestData()
+        {
+            yield return new object[] { "  ", null, new Attribute[0] };
+            yield return new object[] { "  ", new Attribute[0], new Attribute[0] };
+
+            Attribute[] attributes1 = new Attribute[] { new MockAttribute1(), new MockAttribute1() };
+            yield return new object[] { "name", attributes1, new Attribute[] { attributes1[0] } };
+
+            Attribute[] attributes2 = new Attribute[] { new MockAttribute1(), new MockAttribute2() };
+            yield return new object[] { "name", attributes2, attributes2 };
+
+            Attribute[] attributes3 = new Attribute[] { null, new MockAttribute1() };
+            yield return new object[] { "name", attributes3, new Attribute[] { attributes3[1] } };
+
+            Attribute[] attributes4 = new Attribute[] { new CustomTypeIdAttribute(1), new CustomTypeIdAttribute(1), new CustomTypeIdAttribute(2), new CustomTypeIdAttribute(null) };
+            yield return new object[] { "name", attributes4, new Attribute[] { attributes4[0], attributes4[2] } };
+        }
+
+        [Theory]
+        [MemberData(nameof(Ctor_String_Attributes_TestData))]
+        public void Ctor_String_Attributes(string name, Attribute[] attributes, Attribute[] expected)
+        {
+            var descriptor = new SubMemberDescriptor(name, attributes);
+            Assert.Equal(expected, descriptor.Attributes.Cast<Attribute>());
+            Assert.Same(descriptor.Attributes, descriptor.Attributes);
```

Comment:
Was your intent to compare the descriptor property against itself?

Reply:
I guess because Same does a ref comparison?

Reply:
nit: there is a lot less occurrences of Assert.Same as opposed to Assert.Equal in the repo (approx 600 vs 4000).
I believe both Same and Equal will always result in the same outcome, so might want to switch to Assert.Equal instead.

cc: @danmosemsft

Reply:
I would use Equal unless you specifically want reference equality. That is unlikely especially for eg strings

Reply:
Use of Same is deliberate across this PR. Making sure the we have reference equality. This is especially important here as if `Attributes` returned a new instance each time, things would be very very bad

Reply:
Fair enough